gpflow/mean_functions.py

```diff
@@ -152,6 +154,46 @@ def __call__(self, X: TensorType) -> tf.Tensor:
         return tf.zeros(output_shape, dtype=X.dtype)
 
 
+class Polynomial(MeanFunction):
+    """
+    A generic polynomial mean function.
+    """
+
+    @check_shapes("w: [broadcast output_dim, broadcast n_terms]")
+    def __init__(
+        self, degree: int, input_dim: int = 1, output_dim: int = 1, w: Optional[TensorType] = None
+    ) -> None:
+        """
+        :param degree: The degree of the polynomial.
+        :param input_dim: Number of inputs / variables this polynomial is defined over.
+        :param output_dim: Number of outputs / polynomials.
+        :param w: Initial weights of the terms of the polynomial.
```

Comment:
If you pass in weights, how should they be ordered, and how many weights do you expect given the `input_dim`, `output_dim` and `degree`? I assume first the constant term, linear terms, etc. Probably best to add this to the docstring.

Reply:
> If you pass in weights, how should they be ordered, and how many weights do you expect given the `input_dim`, `output_dim` and `degree`? I assume first the constant term, linear terms, etc. Probably best to add this to the docstring.

I've now tried explaining this, though it's annoyingly difficult for the multivariate case.